### PR TITLE
treewide: remove `inherit (darwin)` in all-packages.nix

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Description of changes

Continuing the work from #205703

Tracking issue #204303

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -110,7 +110,7 @@ with pkgs;
     by Hydra.
   '';
 
-  tests = callPackages ../test {};
+  tests = callPackages ../test { };
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix]: Done

**Review by pennae — CHANGES_REQUESTED** [review]
the actual inherit moves look good, but there are a lot more things going on here. would be nice to split them out into their own commits (or even their own PR, but we won't insist)

File: pkgs/applications/misc/binocle/default.nix [hunk 2]
```diff
@@ -3,17 +3,16 @@
...
-, xorg
 , vulkan-loader
+, darwin
+, xorg
 }:
 
+let
+  inherit (darwin.apple_sdk.frameworks)
+    AppKit CoreFoundation CoreGraphics CoreVideo Foundation Metal QuartzCore;
+  inherit (xorg) libX11 libXcursor libXi libXrandr;
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/misc/binocle/default.nix:14]: The reason behind this change is the same. 

Nonetheless it deserves some more granular treewide change. I will split each package in its own commit.

**Comment by winterqt** [comment]
Why not force push if you're going to continue these changes? (Admittedly not sure if we even want to, it's somewhat up in the air and we probably don't want to do this treewide...)

**Comment by AndersonTorres** [comment]
There are too much things to be done on this. Also I don't use Apple tools, so I can't take too much responsibility on this.

**Comment by winterqt** [comment]
Fair enough.

A better comment than "nuke and restart@ could have been said, though, as it implies you were going to start the effort again.

**Pull request closed without merging**